Golem.Tests/StartStopTests.cs

```diff
@@ -52,20 +52,20 @@ public async Task StartStop_VerifyStatusAsync()
             string golemPath = await PackageBuilder.BuildTestDirectory();
             _output.WriteLine("Path: " + golemPath);
 
-            var golem = await TestUtils.Golem(golemPath, loggerFactory);
+            await using var golem = (Golem)await TestUtils.Golem(golemPath, loggerFactory);
 
-            var status = new PropertyChangedHandler<Golem, GolemStatus>(nameof(IGolem.Status), loggerFactory).Observe(golem);
+            var status = TestUtils.StatusChannel(golem, loggerFactory);
 
             var startTask = golem.Start();
-            Assert.Equal(GolemStatus.Starting, status.Value);
+            Assert.Equal(GolemStatus.Starting, await TestUtils.ReadChannel<GolemStatus>(status));
```

Comment:
Every single use of `PropertyChangeHandler` got replace with reading a channel to avoid a race condition.